source/game.c

```diff
@@ -2203,287 +2207,349 @@ static bool check_and_score_joker_for_event(ListItr* starting_joker_itr, CardObj
     return false;
 }
 
-static void played_cards_update_loop(bool* discarded_card, bool* sound_played)
+
+static void play_starting_played_cards_update(int played_idx)
 {
-    // TODO: Break this function up into smaller ones.
+    if (played_idx == played_top && (timer % FRAMES(10) == 0 || !card_object_is_selected(played[played_top - scored_card_index])) && timer > FRAMES(40))
+    {
+        scored_card_index--;
 
-    // So this one is a bit fucking weird because I have to work kinda backwards for everything because of the order of the pushed cards from the hand to the play stack
-    // (also crazy that the company that published Balatro is called "Playstack" and this is a play stack, but I digress)
-    for (int i = 0; i <= played_top; i++)
+        if (scored_card_index == 0)
+        {
+            _joker_scored_itr = list_itr_create(&_owned_jokers_list);
+            timer = TM_ZERO;
+            play_state = PLAY_BEFORE_SCORING;
+        }
+    }
+
+    played[played_idx]->sprite_object->tx = int2fx(HAND_PLAY_POS.x) + (int2fx(played_top - played_idx) - int2fx(played_top) / 2) * -27;
+    played[played_idx]->sprite_object->ty = int2fx(HAND_PLAY_POS.y);
+
+    if (card_object_is_selected(played[played_idx]) && played_top - played_idx >= scored_card_index)
+    {
+        played[played_idx]->sprite_object->ty -= int2fx(10);
+    }
+}
+
+// returns true if the scoring loop has returned early
+static bool play_before_scoring_cards_update(int played_idx)
```

Comment:
```suggestion
static bool play_before_scoring_cards_update()
```

The index is not needed here too.